x-pack/plugins/apm/server/lib/search_strategies/correlations/queries/query_field_value_pairs.ts

```diff
@@ -48,17 +48,17 @@ export const fetchTransactionDurationFieldValuePairs = async (
   esClient: ElasticsearchClient,
   params: SearchServiceFetchParams,
   fieldCandidates: Field[],
-  progress: AsyncSearchProviderProgress
+  state: AsyncSearchServiceState
 ): Promise<FieldValuePairs> => {
   const fieldValuePairs: FieldValuePairs = [];
 
   let fieldValuePairsProgress = 1;
 
   for (let i = 0; i < fieldCandidates.length; i++) {
```

Comment:
Perhaps this would become easier to read if you create a utility function that makes it explicit that we want to iterate sequentially through the array (as opposed to doing it in parallel with `arr.map`):

```js
async function runInSequence(items, fn) {
  const results = [];

  for (const item of items) {
    results.push(await fn(item));
  }

  return results;
}
```

```js
const fieldValuePairs = runInSequence(fieldCandidates, async (fieldName) => {
	
})
```

Reply:
Good idea, refactored in dd7636c.